### PR TITLE
Added nFirstBwtPos in CCoins

### PR DESCRIPTION
**Pull request by abi87**
The attribute nFirstBwtPos is used only for certificates, while it is set to -1 for coins from ordinary transactions.
Removal of IsFromBackwardTransfer for CTxOut is ongoing and will be subject of different pull request

### CODE REVIEW

File: src/coins.cpp [hunk 1]
```diff
@@ -51,13 +53,21 @@ void CCoins::From(const CScCertificate &cert, int nHeightIn, int bwtMaturityHeig
     nVersion           = cert.nVersion;
     nBwtMaturityHeight = bwtMaturityHeight;
     ClearUnspendable();
+    nFirstBwtPos = vout.size();
+    for(int idx = 0; idx < this->vout.size(); ++idx) {
```
Comment (alsala) [inline src/coins.cpp]: idx here can be a size_t type

File: src/coins.h [hunk 2]
```diff
@@ -97,6 +97,11 @@ class CCoins
     //! as new tx version will probably only be introduced at certain heights
     int nVersion;
 
+    //| If coins comes from a certificate, nFirstBwtPos represents the position of the first backward transfer;
+    //! All outputs after nFirstBwtPos, including nFirstBwtPos, are backward transfers
+    //! If coins comes from a tx, it is currently set to zero and not used
```
Comment (alsala) [inline src/coins.h]: Fix the comment; moreover a static const in NOT_VALID_POS = -1 could be used

**Review by alsala — APPROVED** [review]

File: src/coins.cpp (hunk 1 above)
Comment (abi87) [reply src/coins.cpp]: Done

File: src/coins.h (hunk 2 above)
Comment (abi87) [reply src/coins.h]: Done

**Pull request merged**